resolver/resolver.go

```diff
@@ -29,9 +29,7 @@ type Resolver struct {
 	rs      *records.RecordGenerator
 	rsLock  sync.RWMutex
 	rng     *rand.Rand
-
-	// pluggable external DNS resolution, mainly for unit testing
-	extResolver exchanger.Exchanger
+	exchanger.Forwarder
```

Comment:
I see where you're going with this. Could also just call the func type
Forward (it totally makes sense since it's a func type, not an interface
type). Then you get readability and drop the added/unneeded complexity.

On Wed, Oct 7, 2015 at 5:40 PM, Tomás Senart <notifications@github.com>
wrote:

> In resolver/resolver.go
> <https://github.com/mesosphere/mesos-dns/pull/307#discussion_r41451364>:
>
> > @@ -29,9 +29,7 @@ type Resolver struct {
> >  	rs      *records.RecordGenerator
> >  	rsLock  sync.RWMutex
> >  	rng     *rand.Rand
> > -
> > -	// pluggable external DNS resolution, mainly for unit testing
> > -	extResolver exchanger.Exchanger
> > +	exchanger.Forwarder
>
> Yeah, in this case that's fine since it's becoming privatised. But when
> your API consists of anonymous functions being passed around, it's useful
> to have a more descriptive utility method attached to it so that you can
> name the functions other things than verbs.
>
> If you look closer, there is a language mismatch. NewForwarder returns a
> Forwarder which is stored in a field with a verb name forward. With the
> utility method, this goes away.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mesosphere/mesos-dns/pull/307/files#r41451364>.
>